src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -29,19 +32,114 @@ case class NoOpFingerprint() {
 }
 
 // IndexLogEntry-specific Content that uses IndexLogEntry-specific fingerprint.
-case class Content(root: String, directories: Seq[Content.Directory])
+case class Content(root: Directory) {
+  @JsonIgnore
+  lazy val files: Seq[Path] = rec(new Path(root.name), root)
+
+  private def rec(prefixPath: Path, directory: Directory): Seq[Path] = {
```

Comment:
rename it to `recursiveFileListing` or something. Also push it inside `lazy val files` definition